system/handlers/admin/datamanager/formbuilder_question.cfc

```diff
@@ -244,6 +244,21 @@ component extends="preside.system.base.AdminHandler" {
 		}
 	}
 
+	private void function preBatchDeleteRecordsAction( event, rc, prc, args={} ) {
+		var records = args.records ?: QueryNew( "" );
+		var logger  = args.logger  ?: NullValue();
+		var index   = 1;
+
+		for ( var record in records ) {
+			if ( formBuilderQuestionService.questionIsInUse( questionId=record.id ) ) {
+				logMessage( logger, "warn", translateResource( uri="preside-objects.formbuilder_question:batchAction.question.in.use.warning", data=[ record.label ] ) );
+
+				QueryDeleteRow( args.records, index );
+			}
+			index++;
```

Comment:
This loop is problematic. If you delete row 1, all the rows shift up, so if you then delete row 2, you've actually deleted the original row 3.

Simplest solution would be to only increase the index if you haven't deleted the row (put it in the else{} ).